dvc/tree/gdrive.py

```diff
@@ -510,35 +511,68 @@ def exists(self, path_info, use_dvcignore=True):
         else:
             return True
 
-    def _list_paths(self, prefix=None):
-        if not self._ids_cache["ids"]:
-            return
+    def ls(
+        self, path_info, detail=False, recursive=False
+    ):  # pylint: disable=arguments-differ
+        if recursive:
+            for entry in self.walk_files(path_info, detail=detail):
+                if detail:
+                    yield entry
+                else:
+                    yield str(entry)
+            return None
+
+        parent_ids = self._path_to_item_ids(path_info.path)
+        if not parent_ids:
+            return None
 
-        if prefix:
-            dir_ids = self._ids_cache["dirs"].get(prefix[:2])
-            if not dir_ids:
-                return
-        else:
-            dir_ids = self._ids_cache["ids"]
         parents_query = " or ".join(
-            f"'{dir_id}' in parents" for dir_id in dir_ids
+            f"'{dir_id}' in parents" for dir_id in parent_ids
         )
         query = f"({parents_query}) and trashed=false"
 
+        parent_path = path_info.path
         for item in self._gdrive_list(query):
-            parent_id = item["parents"][0]["id"]
-            yield posixpath.join(
-                self._ids_cache["ids"][parent_id], item["title"]
-            )
+            item_path = posixpath.join(parent_path, item["title"])
+            if detail:
+                if item["mimeType"] == FOLDER_MIME_TYPE:
+                    yield {"type": "directory", "name": item_path}
+                else:
+                    yield {
+                        "type": "file",
+                        "name": item_path,
+                        "size": item["fileSize"],
+                        "checksum": item["md5Checksum"],
+                    }
+            else:
+                yield item_path
+
+        if parent_ids:
```

Comment:
since we cache only `min()` next time this call can return a different result?

Reply:
No, not really. I can manually verify that caching works, and also this usage is something that I borrowed from the code above. Though I am not quite sure about why and how (probably the parent_ids is always 1-length list? maybe not but it seems that way)